framework/src/wifi_manager/wifi_manager_dhcp_internal.c

```diff
@@ -0,0 +1,209 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include "wifi_manager_dhcp_internal.h"
+
+#ifdef CONFIG_WIFIMGR_ERROR_REPORT
+#include <error_report/error_report.h>
+#define WIFIADD_ERR_RECORD(reason_code)	ERR_DATA_CREATE(ERRMOD_WIFI_MANAGER_DHCP, reason_code)
+#else
+#define WIFIADD_ERR_RECORD(reason_code)
+#endif
+/**
+ * Enum, definitions and variables
+ */
+#define WIFIMGR_SOFTAP_IFNAME    CONFIG_WIFIMGR_SOFTAP_IFNAME
+#define WIFIMGR_STA_IFNAME       CONFIG_WIFIMGR_STA_IFNAME
+#define WIFIMGR_SET_IP4ADDR(intf, ip, netmask, gateway)         \
+	do {                                                        \
+		int res = -1;                                           \
+		res = netlib_set_ipv4addr(intf, &ip);                   \
+		if (res == -1) {                                        \
+			ndbg("[WM] set ipv4 addr error\n");                \
+		}                                                       \
+		res = netlib_set_ipv4netmask(intf, &netmask);           \
+		if (res == -1) {                                        \
+			ndbg("[WM] set netmask addr error\n");             \
+		}                                                       \
+		res = netlib_set_dripv4addr(intf, &gateway);            \
+		if (res == -1) {                                        \
+			ndbg("[WM] set route addr error\n");               \
+		}                                                       \
+	} while (0)
+
+#ifndef CONFIG_WIFIMGR_DISABLE_DHCPS
+static dhcp_node_s g_dhcp_list = {WIFIMGR_IP4_ZERO, WIFIMGR_MAC_ZERO};
+#endif
+static uint8_t num_sta = 0;
+
+/**
+ * Internal DHCP server APIs
+ */
+void dhcps_inc_num(void)
+{
+	num_sta++;
+}
+
+void dhcps_dec_num(void)
+{
+	if (num_sta > 0) {
+		num_sta--;
+	}
+}
+
+void dhcps_reset_num(void)
+{
+	num_sta = 0;
+}
+
+uint8_t dhcps_get_num(void)
+{
+	return num_sta;
+}
+
+#ifndef CONFIG_WIFIMGR_DISABLE_DHCPS
+wifi_manager_result_e dhcps_start(dhcp_sta_joined cb)
+{
+	struct in_addr ip = {.s_addr = 0x012fa8c0 };
```

Comment:
Is it magic number or generic number?

Reply:
It means '192.168.47.1' normally used ip address for softap. I attached a comment. Thanks!